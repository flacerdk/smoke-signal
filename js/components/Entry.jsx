import React from 'react'

const Entry = (props) => {
  const createMarkup = () => ({ __html: props.text })
  // Trusting that feedparser does proper sanitization here.
  return (
    <div id="active_entry">
      <a className="entry_title" href={props.url}>{props.title}</a>
      <div className="entry_body" dangerouslySetInnerHTML={createMarkup()} />
    </div>
  )
}

Entry.propTypes = {
  title: React.PropTypes.string,
  text: React.PropTypes.string,
  url: React.PropTypes.string,
}

module.exports = Entry
