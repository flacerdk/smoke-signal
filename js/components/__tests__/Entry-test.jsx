import React from 'react'
import Entry from '../Entry'
import renderer from 'react-test-renderer'

describe('Entry', () => {
  it('has read class when read', () => {
    const component = renderer.create(
      <Entry
        text="Test text"
        title="Test title"
        url="http://example.com/test_url"
        read
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('has unread class when unread', () => {
    const component = renderer.create(
      <Entry
        text="Test text"
        title="Test title"
        url="http://example.com/test_url"
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
