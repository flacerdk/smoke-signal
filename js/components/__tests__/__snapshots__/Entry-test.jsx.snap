exports[`Entry has read class when read 1`] = `
<div
  className="entry read">
  <a
    className="entry_title"
    href="http://example.com/test_url">
    Test title
  </a>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test text",
      }
    } />
</div>
`;

exports[`Entry has unread class when unread 1`] = `
<div
  className="entry unread">
  <a
    className="entry_title"
    href="http://example.com/test_url">
    Test title
  </a>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "Test text",
      }
    } />
</div>
`;
